### ISSUE TITLE
LiveReload works for root.tsx but not routes/_index.tsx

### ISSUE THREAD

**Issue by TheHolyWaffle** [remix-run/remix#7899]
### What version of Remix are you using?

2.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Minimal reproduction: https://github.com/TheHolyWaffle/remix-live-reload-7899-reproduction

Manual reproduction:

1. Follow the quickstart instructions (incl setting up `<LiveReload />`): https://remix.run/docs/en/main/start/quickstart
2. Create the file `<project>/app/routes/_index.tsx`:
```
export default function() {
    return <div>Index 1</div>
}
```
3. `npx remix dev`
4. Open `http://localhost:3000/` in browser
5. In `_index.ts`, change the `1` to `2`

### Expected Behavior

The browser shows "Index 2"

### Actual Behavior

The browser still shows "Index 1". Only after refreshing it shows "Index 2"

`npx remix dev` output:
```
 info  rebuilding... (~ app/routes/_index.tsx)
 info  rebuilt (71ms)
 info  app server ready (85ms)
 info  hmr
```

Browser console output:
```
[remix] file changed: app/routes/_index.tsx
[remix] rebuilt (74ms)
[HMR] Component changed [app/routes/_index.tsx]
[HMR] Update accepted by remix:manifest
```

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2. If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.

We plan to continue to address 2 types of issues in Remix v2:

- Bugs that pose security concerns
- Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

- If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
- If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
